tests/test_url_handler.py

```diff
@@ -0,0 +1,700 @@
+    }
+    project = factories.ProjectFactory()
+    # Create the surts we're expecting to see represented in the returned dict.
+    [factories.SURTFactory(url_project=project, value=surts[key][0]) for key in iter(surts)]
+    expected = {'org': [(x, surts[x][1] if surts.get(x) else None) for x in alnum_list]}
+    # Create another unrelated SURT to make sure we aren't grabbing everything.
+    factories.SURTFactory()
+    returned = url_handler.alphabetical_browse(project)
+
+    assert returned == expected
+
+
+@pytest.mark.parametrize('surt, expected', [
+    ('http://(,)', {}),
+    ('http://(org,)', {'org': [(x, None) for x in alnum_list]})
+])
+def test_alphabetical_browse_domains_not_found(surt, expected):
+    project = factories.ProjectFactory()
+    factories.SURTFactory(url_project=project, value=surt)
+    returned = url_handler.alphabetical_browse(project)
+
+    assert returned == expected
+
+
+def test_alphabetical_browse_fake_project():
+    project = 'not a real project'
+    with pytest.raises(http.Http404):
+        url_handler.alphabetical_browse(project)
+
+
+def test_get_metadata():
+    project = factories.ProjectFactory()
+    vals = [x.metadata for x in factories.MetadataValuesFactory.create_batch(3)]
+    metadata = [factories.ProjectMetadataFactory(project=project, metadata=x) for x in vals]
+    expected = [(x, [models.Metadata_Values.objects.get(metadata=x).value]) for x in metadata]
+    returned = url_handler.get_metadata(project)
+
+    for each in returned:
+        assert str(each) in str(expected)
+
+
+def test_get_metadata_with_valueset():
+    project = factories.ProjectFactory()
+    valset = factories.ValuesetFactory()
+    vals = [factories.ValuesetValuesFactory(valueset=valset).value for i in range(3)]
+    metadata = factories.MetadataFactory(value_sets=[valset])
+    factories.ProjectMetadataFactory(project=project, metadata=metadata)
+    returned = url_handler.get_metadata(project)
+
+    for _, value_iter in returned:
+        for value in value_iter:
+            assert value in vals
+
+
+def test_handle_metadata(rf):
+    request = rf.post(
+        '/',
+        {
+            'color': ['blue', 'other_specify'],
+            'group': ['alpha', 'other_specify'],
+            'name': ['Dan', 'Sara'],
+            'title': ['other_specify'],
+            'job': ['other_specify'],
+        }
+    )
+    posted_data = {
+        'color': ['blue', 'other_specify'],
+        'color_other': 'magenta',
+        'group': ['alpha', 'other_specify'],
+        'name': ['Dan', 'Sara'],
+        'title': 'other_specify',
+        'title_other': 'Monsieur',
+        'job': 'other_specify',
+    }
+    expected = {
+        'color': ['blue', 'magenta'],
+        'color_other': 'magenta',
+        'group': ['alpha'],
+        'name': ['Dan', 'Sara'],
+        'title': 'Monsieur',
+        'title_other': 'Monsieur',
+        # Is this really what we want returned if the 'other' attribute isn't set?
```

Comment:
ok, I'll add an issue to track this. do you think I should change the expected value and make this an expected failure on top of creating the issue?